### PR TITLE
check $?.success? after waitpid

### LINKED ISSUE #144: doesn't check $?.exitstatus after waitpid

### PR DESCRIPTION
**Pull request by obfusk**
## 🔤 Polyglot PR

Fixes #144.

## Type of change

- [ ] Docs update (changes to the readme or a site page, no code changes)
- [ ] Ops wrangling (automation or test improvements)
- [x] Bug fix (non-breaking change that fixes an issue)
- [ ] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Sweet release (needs a lot of work and effort)
- [ ] Something else (explain please)

### Checklists

- [ ] If modifying code, at least one test has been added to the suite

### CODE REVIEW

**Comment by obfusk** [comment]
cc @eighthave

File: lib/jekyll/polyglot/patches/jekyll/site.rb [hunk 1]
```diff
@@ -43,7 +43,9 @@ def process
             while pids.length >= (lang == all_langs[-1] ? 1 : nproc)
               sleep 0.1
               pids.map do |lang, pid|
-                pids.delete lang if waitpid pid, Process::WNOHANG
+                next unless waitpid pid, Process::WNOHANG
+                pids.delete lang
+                raise Errors::FatalException.new "subprocess #{pid} (#{lang}) failed (#{$?.exitstatus})" unless $?.success?
```
Comment (untra) [inline lib/jekyll/polyglot/patches/jekyll/site.rb:48]: I'm learning some new ruby, `$?` is crafty

Comment (untra) [reply lib/jekyll/polyglot/patches/jekyll/site.rb:48]: can we get a test for this? something that checks a site build will raise `FatalException` if a build fails. let me know if you need further assistance.

Comment (obfusk) [reply lib/jekyll/polyglot/patches/jekyll/site.rb:48]: I'll look at writing a test when I have some time :)

**Pull request closed without merging**